src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -49,7 +52,12 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
 
             ConnectionId = GenerateConnectionId(Interlocked.Increment(ref _lastConnectionId));
 
-            _rawSocketInput = new SocketInput(Memory, ThreadPool);
+            if (ServerOptions.MaxInputBufferLength.HasValue)
```

Comment:
The default is 1MB, user can set any value between 1 byte and Int32.MaxValue, plus `null` to completely disable all code related to this change, for both perf and also in case this code has a bug.